lib/asciidoctor/converter/semantic-html5.rb

```diff
@@ -53,10 +69,52 @@ def generate_document_title node
     end
   end
 
+  def generate_authors node
+    return if node.authors.empty?
+
+    if node.authors.length == 1
+      %(<p class="byline">
+#{format_author node, node.authors.first, 1}
+</p>)
+    else
+      result = ['<ul class="byline">']
+      node.authors.each_with_index do |author, index|
+        result << "<li>#{format_author node, author, index + 1}</li>"
+      end
+      result << '</ul>'
+      result.join LF
+    end
+  end
+
+  def format_author node, author, index
+    with_context 'author' do
+      %(<span class="author" data-index="#{index}">#{node.sub_replacements author.name}#{author.email ? %( #{node.sub_macros author.email}) : ''}</span>)
+    end
+  end
+
+  def with_context name, &block
+    (@convert_context ||= []).push name
```

Comment:
I'm using an array in case we need nested context. Arguably, we could use a string...?

Reply:
The array seems reasonable here. I may rename this to in_context rather than with_context. You are not adding more context (variables) to the scope, which is usually what with_context means. in_context implies that you are inside some sort of context (but not contributing to it), which is more appropriate.